controllers/storagecluster/external_resources.go

```diff
@@ -254,6 +254,8 @@ func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, inst
 			return r.onboardConsumer(instance, externalClusterClient)
 		} else if instance.Status.Phase == statusutil.PhaseOnboarding {
 			return r.acknowledgeOnboarding(instance, externalClusterClient)
+		} else if instance.Status.Phase == statusutil.PhaseReady {
+			return r.enableReportStatusCronJob(instance, externalClusterClient)
```

Comment:
Why do we need to run this only when we are Ready? why can't we just call this in the else section. 
Also, this change means that the code below (the next `else if`)will not apply to Ready state anymore - which I don't think was your original intention.